controllers/tykoasapidefinition_controller.go

```diff
@@ -188,6 +224,192 @@ func (r *TykOasApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 	return ctrl.Result{RequeueAfter: reqA}, err
 }
 
+// reconcileVersionedOasApi checks if a versioned API can be deleted and isn't a dangling Versioned API
+// i.e an API that was previously part of the version list of a Base API.
+func (r *TykOasApiDefinitionReconciler) reconcileVersionedOasApi(
+	ctx context.Context,
+	log logr.Logger,
+	versionedTykOAS *v1alpha1.TykOasApiDefinition,
+) error {
+	if err := r.cleanUpDanglingVersionedOasAPIs(ctx, versionedTykOAS, log); err != nil {
+		return err
+	}
+
+	if !versionedTykOAS.ObjectMeta.DeletionTimestamp.IsZero() {
+		if versionedTykOAS.Status.GetIsDefaultVersion() {
+			errMsg := "Cannot delete default API version. Remove it from base API's default first"
+
+			versionedTykOAS.Status.LatestTransaction = v1alpha1.TransactionInfo{
+				Time:   metav1.Now(),
+				Status: v1alpha1.TransactionStatus(errMsg),
```

Comment:
Status must be [`Failed`](https://github.com/TykTechnologies/tyk-operator/blob/5763498f755e8b8f1a8283616a53ae4dc183f2e3/api/v1alpha1/common.go#L5-L23), whereas the error message can be specified in `Error` field of TransactionInfo struct, [here](https://github.com/TykTechnologies/tyk-operator/blob/5763498f755e8b8f1a8283616a53ae4dc183f2e3/api/v1alpha1/common.go#L33-L34).